umm-lib/src/cmr/umm/iso_smap/granule/spatial.clj

```diff
@@ -16,20 +16,48 @@
   [spatial-map]
   (some #{:ascending-crossing} (keys spatial-map)))
 
+(defn- orbit-calculated-spatial-domain?
+  "Returns true or false depending on the existence of the orbit-cal-spatial-domain keys in the
+  spatial coverage map."
+  [spatial-map]
+  (let [spatial-map-keys (keys spatial-map)]
+    (or (some #{:orbital-mode-name} spatial-map-keys)
+        (some #{:orbit-number} spatial-map-keys)
+        (some #{:start-orbit-number} spatial-map-keys)
+        (some #{:stop-orbit-number} spatial-map-keys)
+        (some #{:equator-crossing-longitude} spatial-map-keys)
+        (some #{:equator-crossing-date-time} spatial-map-keys))))
+
 (defn xml-elem->SpatialCoverage
   "Returns a UMM SpatialCoverage from a parsed XML structure"
   [xml-struct]
   (let [spatial-elems (cx/elements-at-path xml-struct extent-path)
         geometries (flatten (map spatial/decode spatial-elems))]
     (when (seq geometries)
       (g/map->SpatialCoverage (util/remove-map-keys empty?
-                                {:geometries (remove orbit? geometries)
+                                {:geometries (remove orbit-calculated-spatial-domain?
```

Comment:
rewrite the geometries value part as: 
```
(->> geometries
         (remove orbit?)
         (remove orbit-calculated-spatial-domain?))
```

Reply:
I was thinking of that but I remember Chris once said sth like we need to have at least three operations on the object when using threading...

Reply:
Yeah 3 lines is generally the rule of thumb. Leo's example is 3 lines and looks good to me.

You would call `(seq collection)` not `(-> collection seq)` for example.